### PR TITLE
v0.21.0 pre

### PR DESCRIPTION
**Pull request by chharvey**
merge branch `v0.21.0-pre` into `master`

### CODE REVIEW

**Comment by johnboxall** [comment]
We'd really love to see an updated  release of Harp with the latest version of Terraform! 

However ... Terraform v1.0.0 introduced a regression to Harp's expected behaviour which is causing tests to fail when using the latest version of Terraform.

In particular, as referenced in https://github.com/sintaxi/harp/pull/526, the Jade's behaviour has changed which is causing a number of tests including https://github.com/sintaxi/harp/blob/master/test/basic.js#L26-L37 to fail.

This particular test is failing because https://github.com/sintaxi/harp/blob/master/test/apps/basic/public/globals.json.jade is being rendered as HTML instead of JSON. Hence the error message about a `<` character in the test failure.

I believe these failures relate to moving from Harp's Jade fork to Jade mainline: https://github.com/sintaxi/terraform/pull/122/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2L32

With any luck I'll be able to get a few minutes in the coming (days, months years) to investigate the issue in Terraform, and make the downstream change... which should allow us to pull in a working version of Terraform and move ahead with this change :)

<3

cc @ jeremywiebe

**Comment by sintaxi** [comment]
Ok. finally able to take a closer look at this.

I had the same assumptions about the `.json` being rendered as HTML instead of JSON. Turns out the issue is that the `.json` file is being wrapped in a `layout` because the conditions for disabling layout changed when dots got introduced into the paths. It turns out its not really a big deal because this only happens when generating `.json` via `.jade`. Ive not just set it to no layout by default when generating `.json`. You can still explicitly set a layout file for this if you want.

Anyway, the work is ongoing but things are finally moving forward.

File: lib/middleware.js [hunk 1]
```diff
@@ -323,15 +323,15 @@ exports.mwl = function(req, rsp, next){
   // `.js` (Browserify) are actually being used to specify
   // source files
 
-  if (['js'].indexOf(ext) === -1) {
```
Comment (chharvey) [inline lib/middleware.js]: `['js'].indexOf(ext) === -1` seems like an unusual way of saying `ext !== 'js'`; is there a particular reason for wording it this way?

Comment (sintaxi) [inline lib/middleware.js]: nope. my guess would be the code evolved into that from something else. good catch. :)

**Pull request merged**